Source/Scene/Primitive.js

```diff
@@ -1808,17 +1921,29 @@ define([
             //>>includeEnd('debug');
             var attributeValue = getAttributeValue(value);
             batchTable.setBatchedAttribute(instanceIndex, attributeIndex, attributeValue);
+            if (name === 'offset') {
+                primitive._recomputeBoundingSpheres = true;
+            }
         };
     }
 
+    var offsetScratch = new Cartesian3();
+
     function createBoundingSphereProperties(primitive, properties, index) {
         properties.boundingSphere = {
             get : function() {
                 var boundingSphere = primitive._instanceBoundingSpheres[index];
-                var modelMatrix = primitive.modelMatrix;
-                if (defined(modelMatrix) && defined(boundingSphere)) {
-                    boundingSphere = BoundingSphere.transform(boundingSphere, modelMatrix);
+                if (defined(boundingSphere)) {
+                    var modelMatrix = primitive.modelMatrix;
+                    var offset = properties.offset;
+                    if (defined(offset)) {
+                        boundingSphere.center = Cartesian3.add(Cartesian3.fromArray(offset, 0, offsetScratch), boundingSphere.center, boundingSphere.center);
```

Comment:
Shouldn't this be the same as https://github.com/AnalyticalGraphicsInc/cesium/pull/6607/files#diff-74b9ecf8fb5af0a22b41c6d17e3a8aadR1892